functions/index.js

```diff
@@ -12,26 +12,36 @@ exports.lineWebhook = functions.https.onRequest((request, response) => {
 })
 
 const eventHandler = (event) => {
-  if (
-    event.type !== 'message' ||
-    event.message.type !== 'text' ||
-    !event.message.text.match('เปรี้ยง')
-  ) {
-    return null
+  if(event.type == 'message' || event.message.type == 'text') {
+    if (event.message.text.match('ห้องเทรน')) {
```

Comment:
I think this is almost working. But there are a couple of issues here

1. The inverse of this logic doesn't follow [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws)
2. There one else condition that doesn't explicitly return (return undefined)

I'd suggest the following changes
```suggestion
  if(event.type == 'message' && event.message.type == 'text' && event.message.text.match('ห้องเทรน')) {
```